src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs

```diff
@@ -0,0 +1,404 @@
+#if OS_WINDOWS
+using System;
+using System.Collections.Generic;
+using System.Security.Principal;
+using Microsoft.Win32;
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    public class AutoLogonRegistryManager
+    {
+        private IWindowsRegistryManager _registryManager;
+        private string _userSecurityId;
+        List<KeyValuePair<WellKnownRegistries, string>> _standardRegistries;
+
+        public AutoLogonRegistryManager(IWindowsRegistryManager regManager, string sid = null)
+        {
+            _registryManager = regManager;
+            _userSecurityId = sid;
+            InitializeStandardRegistrySettings();
+        }
+
+        private void InitializeStandardRegistrySettings()
+        {
+            _standardRegistries = new List<KeyValuePair<WellKnownRegistries, string>>()
+            {
+                new KeyValuePair<WellKnownRegistries, string>(WellKnownRegistries.ScreenSaver, "0"),
+                new KeyValuePair<WellKnownRegistries, string>(WellKnownRegistries.ScreenSaverDomainPolicy, "0"),
+                new KeyValuePair<WellKnownRegistries, string>(WellKnownRegistries.ShutdownReason, "0"),
+                new KeyValuePair<WellKnownRegistries, string>(WellKnownRegistries.ShutdownReasonUI, "0"),
+                new KeyValuePair<WellKnownRegistries, string>(WellKnownRegistries.LegalNoticeCaption, ""),
+                new KeyValuePair<WellKnownRegistries, string>(WellKnownRegistries.LegalNoticeText, "")
+            };
+        }
+
+        public bool ValidateIfRegistryExistsForTheUser(string sid)
```

Comment:
Doesn't the term "validate" imply an exception will be thrown if validation fails? However, this method returns a bool.